dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,529 @@
+    {
+        // Verify.
+        Verify.NotNull(database);
+        Verify.NotNullOrWhiteSpace(collectionName);
+
+        // Assign.
+        this._database = database;
+        this.CollectionName = collectionName;
+        this._options = options ?? new();
+        this._vectorStoreRecordDefinition = this._options.VectorStoreRecordDefinition ?? VectorStoreRecordPropertyReader.CreateVectorStoreRecordDefinitionFromType(typeof(TRecord), true);
+        var jsonSerializerOptions = this._options.JsonSerializerOptions ?? JsonSerializerOptions.Default;
+
+        // Validate property types.
+        var properties = VectorStoreRecordPropertyReader.SplitDefinitionAndVerify(typeof(TRecord).Name, this._vectorStoreRecordDefinition, supportsMultipleVectors: true, requiresAtLeastOneVector: false);
+        VectorStoreRecordPropertyReader.VerifyPropertyTypes([properties.KeyProperty], s_supportedKeyTypes, "Key");
+        VectorStoreRecordPropertyReader.VerifyPropertyTypes(properties.DataProperties, s_supportedDataTypes, "Data", supportEnumerable: true);
+        VectorStoreRecordPropertyReader.VerifyPropertyTypes(properties.VectorProperties, s_supportedVectorTypes, "Vector");
+
+        // Get storage names and store for later use.
+        this._keyProperty = properties.KeyProperty;
+        this._storagePropertyNames = VectorStoreRecordPropertyReader.BuildPropertyNameToJsonPropertyNameMap(properties, typeof(TRecord), jsonSerializerOptions);
+
+        // Assign mapper.
+        this._mapper = this._options.JsonObjectCustomMapper ??
+            new AzureCosmosDBNoSQLVectorStoreRecordMapper<TRecord>(
+                this._storagePropertyNames[this._keyProperty.DataModelPropertyName],
+                this._storagePropertyNames,
+                jsonSerializerOptions);
+
+        // Use Azure CosmosDB NoSQL reserved key property name as storage key property name.
+        this._storagePropertyNames[this._keyProperty.DataModelPropertyName] = AzureCosmosDBNoSQLConstants.ReservedKeyPropertyName;
+        this._keyStoragePropertyName = AzureCosmosDBNoSQLConstants.ReservedKeyPropertyName;
+
+        // If partition key is not provided, use key property as a partition key.
+        var partitionKeyPropertyName = !string.IsNullOrWhiteSpace(this._options.PartitionKeyPropertyName) ?
+            this._options.PartitionKeyPropertyName! :
+            this._keyProperty.DataModelPropertyName;
+
+        this._partitionKeyStoragePropertyName = this._storagePropertyNames[partitionKeyPropertyName];
+
+        this._nonVectorStoragePropertyNames = properties.DataProperties
+            .Cast<VectorStoreRecordProperty>()
+            .Concat([properties.KeyProperty])
+            .Select(x => this._storagePropertyNames[x.DataModelPropertyName])
+            .ToList();
+    }
+
+    /// <inheritdoc />
+    public Task<bool> CollectionExistsAsync(CancellationToken cancellationToken = default)
+    {
+        return this.RunOperationAsync("CollectionExists", async () =>
```

Comment:
In other places I've used the internal database operation as the operation name, e.g. `GetContainerQueryIterator` in this case.  I figured that the user already knows they are calling CollectionExists, but the internal method call may be useful.
I don't feel super strongly about it though.

Reply:
I followed the same pattern in other methods, but in this case, I wasn't sure if `GetContainerQueryIterator` operation name will be intuitive enough. But I think it will be nice to keep it consistent with other methods, so I renamed it to `GetContainerQueryIterator`.